### PR TITLE
wallet: simplify nTimeSmart calculation

### LINKED ISSUE #2130: Wrong transaction dates for imported private keys
Dear Gridcoin wallet devs,

I'm importing a few private keys / addresses into a new wallet.

Importing the first private key (here: of the beacon, let's call it _'pk1'_) into a new empty wallet works well regarding the restore of all "transactions". But as soon as I import a second private key (let's call it _'pk2'_), all its transactions (after the rescan is complete) get the same date as the most recent 'transaction' related to _pk1_ (here: the last time it rained on the beacon address). After some transactions occurred on an address of the new wallet (let's call it _'add0'_), I restored a copy of the new wallet (its state: with imported _pk1_, but before I had imported _pk2_), imported _pk2_ again, and: all transactions related to _pk2_ get the date of the most recent transaction (now: of _add0_).

So I believe it's a general issue and not related to the initial beacon address/private key import. Or is it?

Thanks in advance for looking at it!

### PR DESCRIPTION
**Pull request by div72**
This PR simplifies the nTimeSmart calculation to use a block's timestamp if the transaction is inside a block and the current time otherwise.
Fixes #2130.

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -483,6 +483,8 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, CWalletDB* pwalletdb)
         CWalletTx& wtx = (*ret.first).second;
```
Comment (jamescowens) [inline src/wallet/wallet.cpp]: I am ok if you want to update the notation for this to modern standards... ret.first->second while you are in here.

**Review by jamescowens — APPROVED** [review]
utACK

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (div72) [reply src/wallet/wallet.cpp]: Done, I could make a PR with a scripted-diff later to update all usages across the codebase if you want.

**Review by cyrossignol — COMMENTED** [review]
I think this change is ok. Reading through all the historical Bitcoin threads and PRs about this... there seems to be a lot of nuances in the algorithm for `nTimeSmart`, and it seems the "smart" time is supposed to help with some legacy use cases and other uncommon situations that Gridcoin doesn't really support. There is no real way to import a standalone transaction, and the RPC/GUI don't report info about unconfirmed transactions that the node didn't send itself (yet).

One edge case handled by the previous version that we may want to reconsider: nodes can create blocks with a timestamp in the future, and this change will cause nodes to store any transactions in those blocks with future timestamp. Apparently, this was especially confusing and one of the main issues mitigated by the smart time.

The PR does solve #2130 on my end.

**Comment by jamescowens** [comment]
Let's go ahead and merge this. If we want to do some fine-tuning. We can do that in another PR.

**Pull request merged**